[JIT] Add torch._C._jit submodule

SplitInfinity opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#52910 [JIT] Add torch._C._jit submodule**

**Summary**
PR #52158 tried to move all JIT bindings from `torch._C` to a new
submodule `torch._C._jit`, but that...did not go well. This pull request
adds the new `torch._C._jit` submodule, but does not migrate the
existing bindings. Instead, it adds a unit test that fails if any new
bindings are added to `torch._C`. A comment in the test instructs
developers to add their new binding to the allowlist if it really should
be in `torch._C`, or to add it to the appropriate submodule (e.g
`torch._C._jit`, for example). The idea is to prevent the issue
described in #51691 from getting *worse* if it cannot be fixed.

**Test Plan**
Continuous integration.

**Fixes**
This commit fixes #51691.

Differential Revision: [D26698373](https://our.internmc.facebook.com/intern/diff/D26698373)

Commit:
[JIT] Add torch._C._jit submodule

**Summary**
PR #52158 tried to move all JIT bindings from `torch._C` to a new
submodule `torch._C._jit`, but that...did not go well. This pull request
adds the new `torch._C._jit` submodule, but does not migrate the
existing bindings. Instead, it adds a unit test that fails if any new
bindings are added to `torch._C`. A comment in the test instructs
developers to add their new binding to the allowlist if it really should
be in `torch._C`, or to add it to the appropriate submodule (e.g
`torch._C._jit`, for example). The idea is to prevent the issue
described in #51691 from getting *worse* if it cannot be fixed.

**Test Plan**
Continuous integration.

**Fixes**
This commit fixes #51691.

[ghstack-poisoned]

SplitInfinity commented on test/test_jit_bindings.py:
```diff
@@ -0,0 +1,194 @@
+import torch
+
+
+class TestBindings(JitTestCase):
```
Will this test actually get run? Do I need to add it to some other file or lists of tests?

albanD replied:
You need to add it to this list: https://github.com/pytorch/pytorch/blob/7a178a8a523d4653a3a2fa10c573b71e7fab1b9a/test/run_test.py#L20

albanD reviewed (approved):
This looks great! Thanks a lot for doing this.

I think this is not actually specific to jit right?
Maybe we can call this `test_public_cpp_bindings.py` ?

But even if you don't have time to do the rename/update the doc, this is completely worth having!

Commit:
Update on "[JIT] Add torch._C._jit submodule"


**Summary**
PR #52158 tried to move all JIT bindings from `torch._C` to a new
submodule `torch._C._jit`, but that...did not go well. This pull request
adds the new `torch._C._jit` submodule, but does not migrate the
existing bindings. Instead, it adds a unit test that fails if any new
bindings are added to `torch._C`. A comment in the test instructs
developers to add their new binding to the allowlist if it really should
be in `torch._C`, or to add it to the appropriate submodule (e.g
`torch._C._jit`, for example). The idea is to prevent the issue
described in #51691 from getting *worse* if it cannot be fixed.

**Test Plan**
Continuous integration.

**Fixes**
This commit fixes #51691.

[ghstack-poisoned]

albanD reviewed (approved):
Thanks for the update. Looks good to go!
